src/test/scala/uk/gov/nationalarchives/dp/client/DataProcessorTest.scala

```diff
@@ -524,6 +531,48 @@ abstract class DataProcessorTest[F[_]](implicit cme: MonadError[F, Throwable]) e
     exception.getMessage should equal(s"Entity type 'StructuralObject' not found for id $id")
   }
 
+  "getEntityXml" should "return the full entity if all fields are provided" in {
+    val id = UUID.randomUUID()
+    val entityResponse = <EntityResponse>
+      <InformationObject>
+        <Title>Title</Title>
+        <Description>A description</Description>
+        <SecurityTag>open</SecurityTag>
+        <Deleted>true</Deleted>
+        <Parent>f567352f-0874-49da-85aa-ac0fbfa3b335</Parent>
+      </InformationObject>
+    </EntityResponse>
+
+    val response = valueFromF(new DataProcessor[F]().getEntityXml(id, entityResponse, InformationObject))
+    response should equal(entityResponse.child(1))
+  }
+
+  "getEntityXml" should "return an error if the entity type in the response doesn't match" in {
+    val id = UUID.randomUUID()
+    val entityResponse = <EntityResponse>
+      <InformationObject>
+      </InformationObject>
+    </EntityResponse>
+
+    val exception = intercept[PreservicaClientException] {
+      valueFromF(new DataProcessor[F]().getEntityXml(id, entityResponse, StructuralObject))
+    }
+
+    exception.getMessage should equal(s"Entity type 'StructuralObject' not found for id $id")
+  }
+
+  "getEntityXml" should "blahreturn an error if the entity type in the response doesn't match" in {
```

Comment:
I'm not sure the test description is quite right :smile:

Reply:
😄 . You know "blahreturn", when you don't really care about what is returned